src/RecipesBase.jl

```diff
@@ -231,15 +237,12 @@ macro recipe(funcexpr::Expr)
     if !(funcexpr.head in (:(=), :function))
         error("Must wrap a valid function call!")
     end
-    if !(isa(func_signature, Expr) && func_signature.head == :call)
-        error("Expected `func_signature = ...` with func_signature as a call Expr... got: $func_signature")
-    end
     if length(func_signature.args) < 2
         error("Missing function arguments... need something to dispatch on!")
     end
 
-    func = get_function_def(func_signature)
     args, kw_body, cleanup_body = create_kw_body(func_signature)
+    func = get_function_def(func_signature, args)
```

Comment:
The drawback is the fool-proof functionanilty regression. But I'm ok with that.

before:
```julia
julia> @recipe a = 1                                                                                                                              
ERROR: Expected `func_signature = ...` with func_signature as a call Expr... got: a                                                                
Stacktrace:                         
 [1] macro expansion at ./REPL.jl:97 [inlined]                           
 [2] (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:73       
```
after:
```julia
julia> @recipe a = 1                
ERROR: type Symbol has no field args                                                                                                              
Stacktrace:                         
 [1] macro expansion at ./REPL.jl:97 [inlined]                           
 [2] (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:73 
```